### ISSUE TITLE
LOD index/mesh not changing with SetLOD()

### ISSUE THREAD

**Issue by Panicat** [SebLague/Solar-System#46]
It stays on the highest resolution. I think I understand how the code generally works for this, selecting an LOD mesh from an array with LOD meshes.

Below, you can see that 3 meshes are in fact generated and put into the array:

![image](https://user-images.githubusercontent.com/59578148/172591650-052535cd-11c1-4de2-8f99-50a79740b600.png)

SetLOD() with index 0, 1, or 2, appears to refresh the mesh in game as it disappears for a very brief moment but it's still at the highest resolution, which is LOD index 0.

The active LOD never changes. I did try to change the SetLOD() function a bit so that it is FORCED to change the active LOD but still seems like it doesn't change:

![image](https://user-images.githubusercontent.com/59578148/172592409-57d73d79-badc-497a-ab69-45033f26489e.png)

The SetLOD() function to show that I have it in its normal state now:

![image](https://user-images.githubusercontent.com/59578148/172591949-36685177-9791-438a-932d-2541b7c30176.png)

If anyone can help it would be appreciated because I need to lower LODs with distance for optimisation reasons. If any more info is required let me know. Thanks.

**Comment by Delofon** [comment]
Check LODHandler.cs. It automatically changes the LOD of every generator in the scene each frame.

**Comment by Panicat** [comment]
I'm not sure what exactly I should do with it...?

I finally tried having a Debug.Log in the SetLOD function to print the index it is about to set and it seems to run twice. One time it actually tries to set it to index 2 which I'm trying, then it does 0 again.

Which also explains something I noticed now, that for a VERY brief moment I see the planet being low resolution before going back to full when this runs. So I guess yes, something does set it to 0 although it may just me a mistake of mine.
